torch/onnx/_internal/fx/exporter.py

```diff
@@ -63,119 +108,45 @@ def _export(
-    and exports it to ONNX by calling `_export`. Notice that ONNX does
-    not represent keyword arguments, so `args` and `kwargs` are normalized by
-    calling `inspect.Signature.bind` and `inspect.BoundArgument.apply_defaults`
-    in the beginning.
-
+    and exports it to ONNX by calling `_export`.
     Args:
         fn: nn.Module or a callable to be exported to ONNX.
         opset_version: the opset version to export the model to. E.g., 14.
         args: the positional arguments to pass to `fn`.
         use_binary_format: whether to return the ONNX model in binary format.
-            If False, `onnx.ModelProto` will be returned. If False, the byte array
+            If False, `onnx.ModelProto` will be returned. If True, the byte array
             generated by `onnx.ModelProto.SerializeToString` is returned.
         kwargs: the keyword arguments to pass to `fn`.
-
     Returns:
         ONNX model in binary format or `onnx.ModelProto`. To select return type,
         use `use_binary_format` argument.
     """
-
-    if isinstance(fn, torch.nn.Module):
-        signature = inspect.signature(fn.forward)
-    else:
-        signature = inspect.signature(fn)
-
-    # We hope the input kwargs will be mapped to bound.args after binding.
-    # If not, we will raise an error.
-    bound = signature.bind(*args, **kwargs)
-    bound.apply_defaults()
-    # keyword-only arguments are not handled.
-    # bound.kwargs only contains keyword-only arguments after calling
-    # bind & apply_defaults, so we throw if it's not empty.
-    assert not bound.kwargs
-
-    class Wrapper(torch.nn.Module):
-        def __init__(self, fn):
-            super().__init__()
-            self.fn = fn
-
-        def forward(self, *args):
-            result, _ = _pytree.tree_flatten(self.fn(*args))
-            return result
-
-    # args will be converted to symbolic tensor. Let's copy to avoid side effects.
-    bound_args = copy.deepcopy(bound.args)
     # Translate callable to FX graph.
-    #
-    # TODO(wechi): There are several symbolic tracing mechanisms to convert
-    # nn.Module to FX graph. We should choose the right one after they are
-    # matured.
-
-    class GraphCaptureCompiler:
-        def __init__(self):
-            self.captured_graph: Optional["torch.fx.GraphModule"] = None
-            self.captured_graph_count = 0
-
-        def compile(self, graph_module: "torch.fx.GraphModule", _):
-            assert self.captured_graph_count == 0
-            self.captured_graph = graph_module
-            self.captured_graph_count += 1
-            return graph_module
-
-    compiler = GraphCaptureCompiler()
-    torch._dynamo.reset()
-    torch._dynamo.optimize(compiler.compile, nopython=True)(Wrapper(fn))(*bound_args)
-    torch._dynamo.reset()
-    assert compiler.captured_graph
+    # TODO(bowbao, titai): Change "real" to "symbolic" after symbolic shape export is supported.
```

Comment:
I think you are referring to the `decomp` pass, where it does `make_fx` inside?
I think for `dynamo.export` traced graph, this is not needed, because it does decomposition for us. For others we do.
I'm adding fx graph `analysis` in follow-up to reflect on when and why a pass is needed. Hope that clarifies things.

Reply:
And do make_fx() and dynamo.export should use the same tracing_mode?

Reply:
Good question. For now I think the answer is no, that particular make_fx should always use "fake".